### PR TITLE
Add env var to skip go-git errors

### LINKED ISSUE #214: Add environment variable to allow git errors and keep processing
Add configuration to be able to ignore errors from go-git and continue iterating the query result.

### PR DESCRIPTION
**Pull request by jfontan**
Fixes #214

### CODE REVIEW

**Comment by erizocosmico** [comment]
`iterator.go` file has a bunch more of possible places where there could be go-git errors

**Comment by jfontan** [comment]
@erizocosmico yes, I'm on it right now

**Comment by jfontan** [comment]
There are no new tests for functions as I could not find a way to make git repository return errors. Deleting files from the path makes it return `ErrObjectNotFound` and it's skipped.

There are also no new tests in `iterator_test.go` `TestRecursiveTreeFileIter` for the same reason.

**Comment by smola** [comment]
@jfontan Maybe you can remove the repository in the middle of an iteration.

**Comment by jfontan** [comment]
I did that, removing index and packfile but the error (`ErrObjectNotFound`) is already skipped by functions.

File: internal/function/history_idx.go [hunk 1]
```diff
@@ -133,6 +184,10 @@ func (f *HistoryIdx) repoHistoryIdx(repo *git.Repository, start, target plumbing
 	}
 
 	if err != nil {
+		log.WithFields(logrus.Fields{
+			"error":  err,
+			"commit": target.String(),
+		}).Error("could not get commit")
```
Comment (erizocosmico) [inline internal/function/history_idx.go]: shouldn't we skip this too if `SkipGitErrors` is `true`?

File: iterator.go [hunk 2]
```diff
@@ -715,11 +771,23 @@ func (i *commitsIter) Close() error {
...
 
+	commits, err := repo.Repo.CommitObjects()
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"repo":  repo.ID,
+			"error": err,
+		}).Error("unable to get commit iterator")
+
+		if session.SkipGitErrors {
+			return nil, err
```
Comment (erizocosmico) [inline iterator.go]: shouldn't we skip this error?

**Pull request merged**